### DISCUSSION TITLE
Training and validation steps does not match for same dataset

### DISCUSSION THREAD

**Discussion by mehadi92** [NVIDIA-NeMo/NeMo#6827] [Q&A] [answered]
Training data = 6344
Validation data = 6344
training batch size = 32
validation batch size 32
GPU = 2

The validation step is `100`. And this is OK because `6344÷32÷2 = ~100
But why training steps is `1100`? Could not get any clue. is it supposed to happen?

```
Epoch 1:  16%|█▋        | 179/1100 [00:45<03:51,  3.97it/s, loss=533, v_num=ting]
Validation DataLoader 0:  59%|█████▉    | 59/100 [00:07<00:05,  7.90it/s]
```

**Comment by mehadi92** [comment]
is that because I add spec augmentation?

**Reply by titu1994** [reply]
That's not just training steps - PTL shows the sum of training and validation steps on that progress bar v

**Reply by mehadi92** [reply]
@titu1994 Could you please give a clue on how 1100 is calculated?

**Reply by mehadi92** [reply]
I think I have found it. I have set `val_check_interval=.10` So validation will be checked 10 times in 1 epoch. 
So, 
```
train step = 100
validation steps = 100 * 10 = 1000
Total steps = 1000 + 100 = 1100
```

**Reply by mehadi92** [reply, marked as answer]
Here is how PyTorch listening calculates the total steps,

[check this ref](https://lightning.ai/docs/pytorch/stable/common/trainer.html#val-check-interval)

```
# Here is the computation to estimate the total number of batches seen within an epoch.

# Find the total number of train batches
total_train_batches = total_train_samples // (train_batch_size * world_size)

# Compute how many times we will call validation during the training loop
val_check_batch = max(1, int(total_train_batches * val_check_interval))
val_checks_per_epoch = total_train_batches / val_check_batch

# Find the total number of validation batches
total_val_batches = total_val_samples // (val_batch_size * world_size)

# Total number of batches run
total_fit_batches = total_train_batches + total_val_batches
```
